### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by garyluu**
Pagination for both the list components

Removes angular-datatables from many places.  Only place left is the listentry component.  Also fix the browse workflow button to go to the right location.  The browse all, browse workflows, and browse tools buttons have been changed the the primary raised material button.  Style should be changed at a later date (Material theming PR) to maintain consistency.

PR #333 for issue ga4gh/dockstore#666

### CODE REVIEW

File: cypress/integration/ToolsSearch.js [hunk 1]
```diff
@@ -1,37 +1,29 @@
-
-describe('Dockstore tool search page', function() {
+describe('Dockstore tool search page', function () {
```
Comment (denis-yuen) [inline cypress/integration/ToolsSearch.js]: `Dockstore tool list page` more appropriate at this time

File: cypress/integration/WorkflowsSearch.js [hunk 2]
```diff
@@ -1,36 +1,29 @@
-describe('Dockstore workflow search page', function() {
+describe('Dockstore workflow search page', function () {
```
Comment (denis-yuen) [inline cypress/integration/WorkflowsSearch.js]: ditto

**Comment by garyluu** [comment]
Note to self: Add x-total-count check in Cypress once the webservice is released

**Comment by garyluu** [comment]
wait till after meeting to review

**Comment by garyluu** [comment]
Changes after meeting:  
- using mostly the 2nd style shown
- readonly docker pull input field
- nothing is ever hidden (except progress bar of course)
- input field for filter is never disabled
- padding up top to prevent bounce-back (and also leaves a bit of room when the input field is clicked)

**Pull request merged**